daemon/daemon.go

```diff
@@ -359,7 +360,21 @@ func (daemon *Daemon) restore() error {
 			if daemon.configStore.AutoRestart && container.shouldRestart() {
 				logrus.Debugf("Starting container %s", container.ID)
 
-				if err := daemon.containerStart(container); err != nil {
+				var retries int
+				for err = daemon.containerStart(container); err != nil && retries <= 5; retries++ {
+					// Best effort to wait for container deps to be up
+					switch e := err.(type) {
+					case errcode.Error:
+						if e.ErrorCode() == derr.ErrorCodeLinkNotRunning {
```

Comment:
I think at one point I did have a util func to avoid making everyone write switch statements but it never made the cut.